src/tools/optimization-options.h

```diff
@@ -328,32 +328,53 @@ struct OptimizationOptions : public ToolOptions {
   bool runningPasses() { return passes.size() > 0; }
 
   void runPasses(Module& wasm) {
-    PassRunner passRunner(&wasm, passOptions);
-    if (debug) {
-      passRunner.setDebug(true);
-    }
-    for (auto& pass : passes) {
-      // We apply the pass's intended opt and shrink levels, if any.
-      auto oldOptimizeLevel = passRunner.options.optimizeLevel;
-      auto oldShrinkLevel = passRunner.options.shrinkLevel;
-      if (pass.optimizeLevel) {
-        passRunner.options.optimizeLevel = *pass.optimizeLevel;
+    std::unique_ptr<PassRunner> passRunner;
+
+    // Flush anything in the current pass runner, and then reset it to a fresh
+    // state so it is ready for new things.
+    auto flushAndReset = [&]() {
+      if (passRunner) {
+        passRunner->run();
+        passRunner.reset();
       }
-      if (pass.shrinkLevel) {
-        passRunner.options.shrinkLevel = *pass.shrinkLevel;
+      passRunner = std::make_unique<PassRunner>(&wasm, passOptions);
```

Comment:
Should this be in an `else`? Otherwise it seems strange to `reset()` the old pass runner just to overwrite it.

Reply:
It's intentional to overwrite - it provides a fresh pass runner, with the right settings etc.

I can remove the `.reset()` though, since `passRunner = ` will effectively do a reset as it overwrites. I did that now.